src/main/java/broker/Log.java

```diff
@@ -0,0 +1,72 @@
+package broker;
+
+import java.util.List;
+
+/**
+ * Log represents the entire log for a single partition in Flux, and more specifically is composed of
+ * multiple LogSegment objects in which each segment represents a chunk of the partition's data on disk.
+ * Log, by extension, is essentially a manager for all these segments.
+ */
+public class Log {
+    private List<LogSegment> segments;
+    private int currentActiveSegmentIdx = 0; // references index in above list
+    private int logStartOffset; // record offsets, not bytes
+    private int logEndOffset;
+    private long currentSizeInBytes;
+
+    // By default, creating a Log will also instantiate 1 empty and mutable LogSegment (default record offset: 0)
+    public Log() {
+        this.segments = List.of(new LogSegment(0, 0));
+        this.logStartOffset = segments.get(0).getStartOffset();
+        this.logEndOffset = segments.get(0).getEndOffset();
+    }
+
+    // Instantiate log with already-made segment
+    public Log(LogSegment segment) {
```

Comment:
I see a `Log` can have multiple segments, so I'm wondering if this constructor could take in a list of segments, or maybe have another constructor that takes in multiple segments. What are your thoughts?

Reply:
yeah good call, will include another constructor